### PR TITLE
Feature/sso signup

### PR DESCRIPTION
**Pull request by deepakprabhakara**
To be merged with calendso/website#47

## What does this PR do?

Updates saml-jackson and supports Google and SAML signup via a redirect from the Cal.com website. It also allows signing in with multiple identities for the same email (only for self-hosted instances)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by PeerRich** [comment]
@zomars @baileypumfleet @emrysal

**Review by zomars — CHANGES_REQUESTED** [review]
Testing the flow [on staging](https://cal.dev) threw this error:

```
[GET] /auth/sso/google?username=zomars001
11:23:46:02
2022-01-25T18:23:46.228Z	561ea629-624d-47f1-94b4-397bf0f85ca8	ERROR	Unhandled Promise Rejection 	{"stack":"Runtime.UnhandledPromiseRejection: ReferenceError: window is not defined\n    at process.<anonymous> (/var/runtime/index.js:35:15)\n    at process.emit (events.js:412:35)\n    at processPromiseRejections (internal/process/promises.js:245:33)\n    at processTicksAndRejections (internal/process/task_queues.js:96:32)","message":"ReferenceError: window is not defined","reason":{"stack":"ReferenceError: window is not defined\n    at _callee5$ (/var/task/node_modules/next-auth/react/index.js:220:160)\n    at tryCatch (/var/task/node_modules/regenerator-runtime/runtime.js:63:40)\n    at Generator.invoke [as _invoke] (/var/task/node_modules/regenerator-runtime/runtime.js:294:22)\n    at Generator.next (/var/task/node_modules/regenerator-runtime/runtime.js:119:21)\n    at asyncGeneratorStep (/var/task/node_modules/next-auth/node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)\n    at _next (/var/task/node_modules/next-auth/node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)\n    at /var/task/node_modules/next-auth/node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7\n    at new Promise (<anonymous>)\n    at /var/task/node_modules/next-auth/node_modules/@babel/runtime/helpers/asyncToGenerator.js:21:12\n    at _signIn (/var/task/node_modules/next-auth/react/index.js:329:18)","message":"window is not defined"},"promise":{}}
Unknown application error occurred
```

This error is coming from here:

https://app.cal.dev/auth/sso/google?username=zomars001

**Comment by PeerRich** [comment]
merged github.com/calcom/website#47

File: components/NavTabs.tsx [hunk 1]
```diff
@@ -17,7 +17,9 @@ const NavTabs: FC<Props> = ({ tabs, linkProps }) => {
   const router = useRouter();
   return (
     <>
-      <nav className="-mb-px flex rtl:space-x-reverse space-x-2 sm:rtl:space-x-reverse space-x-5" aria-label="Tabs">
+      <nav
+        className="-mb-px flex rtl:space-x-reverse space-x-2 sm:rtl:space-x-reverse space-x-5"
```
Comment (alishaz-polymath) [inline components/NavTabs.tsx:21]: `space-x-2` and `space-x-5` together don't make sense. Are you sure this is intended?

Comment (deepakprabhakara) [reply components/NavTabs.tsx:21]: Odd, this was a merge from main. Let me check.

Comment (deepakprabhakara) [reply components/NavTabs.tsx:21]: @alishaz-polymath It's the same as main, looks like some minor formatting difference - https://github.com/calcom/cal.com/blob/main/components/NavTabs.tsx#L20

**Comment by alishaz-polymath** [comment]
Tested Google, works like a charm :)
Testing for SAML remaining...

**Review by pumfleet — COMMENTED** [review]
Code looks good to me, will test shortly

**Comment by alishaz-polymath** [comment]
> Tested Google, works like a charm :) Testing for SAML remaining...
So, while testing for SAML, I noticed that the save button doesn't follow the Calcom design. It would need to be updated
[ref:https://www.loom.com/share/e05d11e95fed4fc8a8f398b34d7dbf5b]

The SAML works when the user doesn't have an account with cal yet, but if they do, it doesn't work currently and asks to use password instead.

Apart from these two, everything looks perf!!

**Review by alishaz-polymath — CHANGES_REQUESTED** [review]
- The save button in `/settings/security` for SAML xml needs visual update
- SAML currently doesn't work for users with existing account, this is not intended.
Everything else works well :)

**Comment by deepakprabhakara** [comment]
Thanks @alishaz-polymath. Save button has been fixed now. For existing accounts I block login from another identity provider if the email has not been verified yet (for security reasons). Otherwise it will go through (just re-tested here).

**Pull request merged**